frontend/components/BookTable.tsx

```diff
@@ -0,0 +1,272 @@
+import React, { useCallback } from "react";
+import { Input, Button, DatePicker } from "antd";
+import moment from "moment";
+import { useMutation, useQuery } from "@apollo/client";
+import { BooksData, Book } from "@/types/book";
+import { ADD_BOOK, GET_BOOKS, UPDATE_BOOK } from "@/utils/graphql";
+import PaginatedTable from "./PaginatedTable";
+import BookForm from "./BookForm";
+
+const { RangePicker } = DatePicker;
+
+const BookTable: React.FC = () => {
+  const { refetch } = useQuery<BooksData>(GET_BOOKS);
+  const [openForm, setOpenForm] = React.useState(false);
+  const [addBook, { data }] = useMutation(ADD_BOOK);
+  const [updateBook, { data: updatedData }] = useMutation(UPDATE_BOOK);
+  const [selectedBook, setSelectedBook] = React.useState<Book | null>(null);
+
+  const fetchData = useCallback(
+    async ({
+      currentPage,
+      currentLimit,
+      filters,
+    }: {
+      currentPage: number;
+      currentLimit: number;
+      filters: Record<string, any>;
+    }) => {
+      const { data } = await refetch({
+        titleFilter: filters.title?.[0] || "",
+        currentPage,
+        currentLimit,
+        startDate: filters.published_date?.[0]?.[0]?.format("YYYY-MM-DD"),
+        endDate: filters.published_date?.[0]?.[1]?.format("YYYY-MM-DD"),
+        authorFilter: filters.author?.[0] || "",
```

Comment:
use of 0 / 1 index, was confusing, what is a better way to handle / architecture this?

Reply:
Could have created variables that's self explanatory